.github/workflows/test.yml

```diff
@@ -0,0 +1,40 @@
+name: Test Server Core
+on: [push, pull_request]
```

Comment:
Edwin and I have been discussing this and we don't think it's necessary to run these tests on pull_request given that we also run them on push. Getting rid of the pull_request run would save resources and help us avoid https://jira.nypl.org/browse/SIMPLY-3592.

Reply:
I guess it depends on the contribution workflow. 

I don't have access to directly push branches into these repos, so my contributions (this one for instance) would only trigger a CI run if it were triggered on a PR.

Reply:
Looking for a solution from github there is a thread about this problem [here](https://github.community/t/how-to-trigger-an-action-on-push-or-pull-request-but-not-both/16662). 

The suggested solution is: 
```yaml
on:
  push:
    branches:
    - develop
  pull_request:
    branches:
    - develop
``` 

Would it work to do something like this? CI always runs on PR, or on any commit to develop branch?

Reply:
I see what you're saying about the permissions, so we need to keep the PR tests. I'd like the tests to run _before_ we push to develop though. Running the test afterwards seems like closing the barn door after the horse has left.

Let's leave it as is for now; we might be able to reduce the number of times we hit Docker, or we can just run the test authenticated as a user with a higher rate limit.

Reply:
Agreed about seeing code tested before it hits the develop branch. 

If the workflow to get code into the develop branch always involves merging a pull request, then I think this solution holds. The code will be tested before its merged by the pull request invocation of the tests, and then tests will be run on the commits to develop after the PR is merged.

The place where tests might not get run is if code is being merged into develop, but not going through the PR process. I don't think this is ever the case, but perhaps some internal NYPL development goes in this way?

Reply:
No, we never push to develop without a PR.

I guess the sticking point for me is, when I push a development branch I want the tests to run while I'm writing the PR, rather than waiting til the PR exists. That way by the time I'm done with the PR I can see the test failures. I don't see a way of doing that without duplicative work, though.